### PR TITLE
:seedling: Some more cleanup for field placement

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Cleanup for fieldplacement in struct

#### What kind of change does this PR introduce?

(Is it a bug fix, feature, docs update, something else?)

- [ ] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the new behavior (if this is a feature change)?**

- [ ] Tests for the changes have been added (for bug fixes/features)

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note

```

### CODE REVIEW

File: checker/check_result.go [hunk 1]
```diff
@@ -56,17 +56,14 @@ const (
 var errSuccessTotal = errors.New("unexpected number of success is higher than total")
 
 // CheckResult captures result from a check run.
-//
-//nolint:govet
 type CheckResult struct {
-	Name    string
-	Version int
```
Comment (spencerschrock) [inline checker/check_result.go:63]: similar comment on human-readability. We should decide if we care about this

File: checker/raw_result.go [hunk 2]
```diff
@@ -25,29 +25,27 @@ import (
...
-//nolint:govet
 type RawResults struct {
-	BinaryArtifactResults       BinaryArtifactData
+	Metadata                    MetadataData
 	BranchProtectionResults     BranchProtectionsData
-	CIIBestPracticesResults     CIIBestPracticesData
-	CITestResults               CITestData
-	CodeReviewResults           CodeReviewData
+	PinningDependenciesResults  PinningDependenciesData
+	SecurityPolicyResults       SecurityPolicyData
```
Comment (spencerschrock) [inline checker/raw_result.go:32]: So I know this was alphabetized recently to find the data easier.
https://github.com/ossf/scorecard/pull/3540

Comment (naveensrinivasan) [reply checker/raw_result.go:32]: https://github.com/ossf/scorecard/pull/3791#discussion_r1456296660

File: checker/check_result.go (hunk 1 above)
Comment (spencerschrock) [reply checker/check_result.go:63]: > So it helps to have performance IMO. Thoughts?

My first thought is, does it matter? I know we have the linter (and could disable it if the answer is no), but is this a performance bottleneck? Compared to the I/O and API operations we need to wait for anyway?

Comment (naveensrinivasan) [reply checker/check_result.go:63]: The compiler utilizes the L1 cache based on the word size.

https://articles.wesionary.team/simple-data-alignment-technique-to-speed-up-your-struct-in-golang-954c43faa1fe
https://www.ardanlabs.com/blog/2013/07/understanding-type-in-go.html

It helps to make it better. 

> I understand API Operations.

**Pull request closed without merging**